src/main/java/com/google/sps/servlets/DashboardServlet.java

```diff
@@ -0,0 +1,75 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.sps.servlets.Property;
+import java.lang.reflect.Field; 
+import java.util.HashMap;
+
+@WebServlet("/dashboard")
+public class DashboardServlet extends HttpServlet {
+
+ /**
+  * This route will return a JSON with the percentages of each interaction
+  */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    try {
+      HashMap<String, Double> interactionPercentages = calculatePercentages();
+      
+      String jsonToSend = convertToJson(interactionPercentages);
+
+      response.setContentType("application/json; charset=UTF-8");
+      response.getWriter().println(jsonToSend);
+    
+    } catch (IllegalAccessException exception) {
```

Comment:
When would this happen?

Reply:
The calculatePercentage() method uses the getDeclared function that could possibly throw an exception. Java wouldn't let me compile if i didn't catch that exception so that's why